### PR TITLE
Add the conservative regridding method

### PR DESCRIPTION
**Pull request by BSchilperoort**
A first conservative regridding method has now been implemented, and a benchmark is included.

TODO's will be made into issues.

### CODE REVIEW

**Comment by geek-yang** [comment]
https://github.com/EXCITED-CO2/xarray-regrid/blob/6681cab20462d2194167d3dfd4e100baa1ed5fbd/src/xarray_regrid/regrid.py#L22

Also need to include "conservative" here, which then becomes `method: Literal["linear", "nearest", "cubic", "conservative"], `

**Comment by BSchilperoort** [comment]
> https://github.com/EXCITED-CO2/xarray-regrid/blob/6681cab20462d2194167d3dfd4e100baa1ed5fbd/src/xarray_regrid/regrid.py#L22
> 
> Also need to include "conservative" here, which then becomes `method: Literal["linear", "nearest", "cubic", "conservative"], `

actually, with the change to `da.regrid.linear()`, this part become obsolete so I won't change it :upside_down_face:

**Comment by geek-yang** [comment]
> > https://github.com/EXCITED-CO2/xarray-regrid/blob/6681cab20462d2194167d3dfd4e100baa1ed5fbd/src/xarray_regrid/regrid.py#L22
> > 
> > Also need to include "conservative" here, which then becomes `method: Literal["linear", "nearest", "cubic", "conservative"], `
> 
> actually, with the change to `da.regrid.linear()`, this part become obsolete so I won't change it 🙃

Oh, yep! Forgot it. Sorry 😂 . You are right haha.

File: src/xarray_regrid/methods.py [hunk 1]
```diff
@@ -25,3 +27,56 @@ def interp_regrid(
...
+    https://discourse.pangeo.io/t/conservative-region-aggregation-with-xarray-geopandas-and-sparse/2715/3
+
+    Args:
+        data: Input dataset.
+        target_ds: Dataset which coordinates the input dataset should be regrid to.
+
+    Returns:
+        Regridded input dataset
+    """
+    coord_names = set(target_ds.coords).intersection(set(data.coords))
```
Comment (geek-yang) [inline src/xarray_regrid/methods.py:52]: Elegant 💡 !

File: benchmarks/benchmarking_conservative.ipynb [hunk 2]
```diff
@@ -0,0 +1,1133 @@
...
+   "source": [
+    "from time import time\n",
+    "import dask.distributed\n",
+    "import xarray_regrid  # Importing this will make Dataset.regrid accessible.\n",
+    "import xarray as xr\n",
+    "from xarray_regrid import Grid, create_regridding_dataset\n",
+    "import xesmf as xe\n",
+    "\n",
+    "client = dask.distributed.Client()\n",
+    "\n",
```
Comment (geek-yang) [inline benchmarks/benchmarking_conservative.ipynb:34]:
```suggestion
```
It can be removed.

**Review by geek-yang — APPROVED** [review]
Super! Nice work!

**Pull request merged**